sdks/python/apache_beam/io/sources_test.py

```diff
@@ -59,28 +59,32 @@ def read(self, range_tracker):
   def split(self, desired_bundle_size, start_position=None, stop_position=None):
     assert start_position is None
     assert stop_position is None
-    with open(self._file_name, 'rb') as f:
-      f.seek(0, os.SEEK_END)
-      size = f.tell()
+    size = self.estimate_size()
 
     bundle_start = 0
     while bundle_start < size:
       bundle_stop = min(bundle_start + LineSource.TEST_BUNDLE_SIZE, size)
-      yield iobase.SourceBundle(1, self, bundle_start, bundle_stop)
+      yield iobase.SourceBundle(bundle_stop - bundle_start,
+                                self,
+                                bundle_start,
+                                bundle_stop)
       bundle_start = bundle_stop
 
   def get_range_tracker(self, start_position, stop_position):
     if start_position is None:
       start_position = 0
     if stop_position is None:
-      with open(self._file_name, 'rb') as f:
-        f.seek(0, os.SEEK_END)
-        stop_position = f.tell()
+      stop_position = self.estimate_size()
```

Comment:
I suggest moving getting the size of the file to a separate method and calling that from both here and estimate_size(). Getting the result of estimate_size() call and setting it as the stop position doesn't look good since stop position has to be precise.